pytket/qir/conversion/conversion.py

```diff
@@ -323,7 +325,8 @@ def _rebase_command_to_gateset(self, command: Command) -> Optional[Circuit]:
         if optype not in self.module.gateset.base_gateset:
             circuit = Circuit(self.circuit.n_qubits, self.circuit.n_bits)
             circuit.add_gate(optype, params, args)
-            self.rebase_to_gateset.apply(circuit)
+            if self.rebase_to_gateset.apply(circuit):
+                raise ValueError("Gate not supported")
```

Comment:
This is a bit weird, attempting a rebase pass and failing if it does anything. Can we just check that the circuit only contains supported gates?

Reply:
Yes I can do that.

Reply:
done in @cqc-melf
[add gateset predicate](https://github.com/CQCL/pytket-qir/pull/42/commits/833d1194d7fd7ffb2443ab8306137e62cb6d4f0e)
[833d119](https://github.com/CQCL/pytket-qir/pull/42/commits/833d1194d7fd7ffb2443ab8306137e62cb6d4f0e)